x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/AnalysisConfig.java

```diff
@@ -69,6 +69,8 @@ public class AnalysisConfig implements ToXContentObject, Writeable {
     public static final String ML_CATEGORY_FIELD = "mlcategory";
     public static final Set<String> AUTO_CREATED_FIELDS = new HashSet<>(Collections.singletonList(ML_CATEGORY_FIELD));
 
+    public static final int MAX_CATEGORIZATION_FIELD_LENGTH = 1001;
```

Comment:
Please add a comment here saying why this is 1001.

Say that the C++ truncates at length 1000 (which is `model::CCategoryExamplesCollector::MAX_EXAMPLE_LENGTH`), and adds an ellipsis on truncation, hence we need to send more than that so this logic still works.

And also say that because we do the tokenization on the Java side now the tokens will still be sent correctly (separately) even if they extend beyond the length of a truncated example.